### PR TITLE
SVA-387 Update theme components and refactor code to match

### PR DESCRIPTION
**Pull request by PammyLo**
- [Description](#description)
  - [Type of change](#type-of-change)
- [Testing locally](#testing-locally)
- [Checklist](#checklist)

# Description

- Throughout the app many different font sizes and paddings/margins were being set. This makes some changes to a few of the Theme components to standardise and avoid need for corrective margins or negative padding. 
   - Biggest impact requiring some margin and padding changes on all screens was removing the default padding from the `Text` component
   - Other change worth noting is introducing a way to easily set the font style following the naming in the Design System in Figma. 
   - Refactored codebase to ensure layout is preserved
 
- Overall there should now be less margins and paddings set throughout the codebase and no more need for negative padding. 

- Also updated the look of the date picker which was missing a few border lines.  

Fixes # (issue number)
[SVA-387](https://sta2020.atlassian.net/browse/SVA-387)

## Type of change

Please delete options that are not relevant

- [X] Non-breaking change which fixes an issue
- [] New Feature (non-breaking change which adds functionality)
- [] Breaking change(fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

# Testing locally

- install npm before starting the app
- start as normal and check the app still looks good 
    - the only pages changed are fixed margins in Profile container and improved look of the calendar date picker (in Register screen)

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have removed any unnecessary comments or console logging
- [x] I have made corresponding changes to the documentation (if required)
- [x] I have addressed accessibility, if needed
- [x] I have followed best practices, e.g. NativeBase approaches and theming
- [x] I have checked the app in dark mode, if making front-end design changes
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have updated the version numbers in `package.json` files in the [app](DEPLOYMENT.md#app-deployment) and/or [api](DEPLOYMENT.md#api-deployment-on-aws) directories as needed

### CODE REVIEW

**Review by M4RC1N76 — APPROVED** [review]
Great work @PammyLo Pamela, I've familiarise myself with the changes and run the App on my android. All good!!

File: app/src/NativeBase/Components/Project/ProjectFullDetails.tsx
Comment (DJMF2015) [inline app/src/NativeBase/Components/Project/ProjectFullDetails.tsx:1]: So far the work is great and good to see it's now possible to 'do away' with negative margining and there are the set paddings and margins laid specified. I also like the fact that you have tidied up some of the layouts to remove unnecessary components e.g.) multiple VStacks. I just have one quick question. I noticed you've added in the 'react-native-vector-icons/MaterialIcons' import from Material Icons. Is this installed as an additional dependency? If so, is it possible to achieve the same result with existing icon libraries installed? For instance 'expo-vector-icons'? I mention this as when i did something similar before the break, Joe was adamant we shouldn't be adding additional dependencies in if at all possible for consistency. ie) we should be using existing expo libraries where possible. I would be keen to know your thoughts! 😊

File: app/src/NativeBase/Theme/StaTheme.tsx
Comment (DJMF2015) [inline app/src/NativeBase/Theme/StaTheme.tsx:1]: Good work!

File: app/src/NativeBase/Components/Project/ProjectFullDetails.tsx
Comment (PammyLo) [reply app/src/NativeBase/Components/Project/ProjectFullDetails.tsx:1]: Hi David, thank you so much for reviewing! You are absolutely right and I agree totally. To put your mind at ease, this is an existing package, it is only imported into this file to use an icon from it. Thanks for keeping us right! :)

Comment (DJMF2015) [reply app/src/NativeBase/Components/Project/ProjectFullDetails.tsx:1]: That's great. Thanks for clarifying this!

**Review by DJMF2015 — APPROVED** [review]
The code changes and amendments are looking good and i am happy to approve your PR. Good work.

**Pull request merged**